rinex/src/observation/record.rs

```diff
@@ -1322,874 +1317,562 @@ impl Decimate for Record {
     }
 }
 
-use crate::algorithm::StatisticalOps;
+#[cfg(feature = "obs")]
 use statrs::statistics::Statistics;
 
-impl Processing for Record {
-    /*
-     * Statistical method wrapper,
-     * applies given statistical function to self (entire record)
-     */
-    fn statistical_ops(
-        &self,
-        ops: StatisticalOps,
-    ) -> (Option<f64>, HashMap<Sv, HashMap<Observable, f64>>) {
-        let mut ret: (Option<f64>, HashMap<Sv, HashMap<Observable, f64>>) = (None, HashMap::new());
-        // eval for clock offsets, if such data exist
-        let clock_offsets: Vec<_> = self
-            .iter()
-            .filter_map(|(_, (clk, _))| {
-                if let Some(clk) = clk {
-                    Some(*clk)
-                } else {
-                    None
-                }
-            })
-            .collect();
-        if clock_offsets.len() > 0 {
-            match ops {
-                StatisticalOps::Max => ret.0 = Some(clock_offsets.max()),
-                StatisticalOps::MaxAbs => ret.0 = Some(clock_offsets.abs_max()),
-                StatisticalOps::Min => ret.0 = Some(clock_offsets.min()),
-                StatisticalOps::MinAbs => ret.0 = Some(clock_offsets.abs_min()),
-                StatisticalOps::Mean => ret.0 = Some(clock_offsets.mean()),
-                StatisticalOps::QuadMean => ret.0 = Some(clock_offsets.quadratic_mean()),
-                StatisticalOps::GeoMean => ret.0 = Some(clock_offsets.geometric_mean()),
-                StatisticalOps::HarmMean => ret.0 = Some(clock_offsets.harmonic_mean()),
-                StatisticalOps::Variance => ret.0 = Some(clock_offsets.variance()),
-                StatisticalOps::StdDev => ret.0 = Some(clock_offsets.std_dev()),
+#[cfg(feature = "obs")]
+use crate::observation::{Observation, StatisticalOps};
+
+#[cfg(feature = "obs")]
+/*
+ * Evaluate specific statistical esimate on this record data set
+ */
+fn statistical_estimate(
+    rec: &Record,
+    ops: StatisticalOps,
+) -> (Option<f64>, HashMap<Sv, HashMap<Observable, f64>>) {
+    let mut ret: (Option<f64>, HashMap<Sv, HashMap<Observable, f64>>) = (None, HashMap::new());
+
+    // Vectorize clock offset (in time), so we can invoke statrs() on it
+    let data: Vec<f64> = rec
+        .iter()
+        .filter_map(|(_, (clk, _))| {
+            if let Some(clk) = clk {
+                Some(*clk)
+            } else {
+                None
             }
+        })
+        .collect();
+
+    if data.len() > 0 {
+        // data exists
+        match ops {
+            StatisticalOps::Max => ret.0 = Some(data.max()),
```

Comment:
Nit: I'd swap these to keep a consistent order (makes it easier to spot copy-paste mistakes).

Reply:
I'm sorry but I don't understand what you mean,   
some pseudo code would help ;)

Reply:
Just swapping the Min and Max lines, since I think that's the order used in other places.